frontend/public/components/modals/alert-routing-modal.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import {Base64} from 'js-base64';
+import {safeDump} from 'js-yaml';
+import * as _ from 'lodash-es';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import {k8sPatch, K8sResourceKind} from '../../module/k8s';
+import {SecretModel} from '../../models';
+
+export const AlertRoutingModal: React.FC<AlertRoutingModalProps> = ({config, secret, cancel, close}) => {
+  const [errorMessage, setErrorMessage] = React.useState();
+  const [inProgress, setInProgress] = React.useState(false);
+
+  React.useEffect(() => {
+    config = _.cloneDeep(config);
+    secret = _.cloneDeep(secret);
+  }, [config, secret]);
+
+  const submit = (event): void => {
+    event.preventDefault();
+
+    _.set(config, ['route', 'group_by'], event.target.elements['input-group-by'].value.split(','));
+    _.set(config, ['route', 'group_wait'], event.target.elements['input-group-wait'].value);
+    _.set(config, ['route', 'group_interval'], event.target.elements['input-group-interval'].value);
+    _.set(config, ['route', 'repeat_interval'], event.target.elements['input-repeat-interval'].value);
+
+    let yamlStringData = '';
+
+    try {
+      yamlStringData = safeDump(config);
+    } catch (e) {
+      setErrorMessage(`Error getting YAML: ${e}`);
+      return;
+    }
+    setInProgress(true);
+    k8sPatch(SecretModel, secret, [{ op: 'replace', path: '/data/alertmanager.yaml', value:  Base64.encode(yamlStringData)}])
+      .then(close, err => {
+        setErrorMessage(err.message);
+        setInProgress(false);
+      });
+  };
+
+  return <form onSubmit={submit} name="form" className="modal-content">
+    <ModalTitle className="modal-header">Edit Routing Configuration</ModalTitle>
+    <ModalBody>
+      <div className="row co-m-form-row">
+        <div className="col-sm-12">
+          <label htmlFor="group-by" className="control-label">
+              Group By
+          </label>
+        </div>
+        <div className="co-m-form-col col-sm-12">
+          <div className="form-inline">
+            <div className="pf-c-input-group">
+              <input placeholder="cluster, alertname"
+                type="text" className="pf-c-form-control"
+                id="input-group-by"
+                defaultValue={_.get(config, ['route', 'group_by'])}
+                aria-describedby="input-group-by-help" />
+            </div>
+          </div>
+        </div>
+      </div>
+      <div className="row co-m-form-row">
+        <div className="col-sm-12">
+          <label htmlFor="group-wait" className="control-label">
+              Group Wait
+          </label>
+        </div>
+        <div className="co-m-form-col col-sm-12">
+          <div className="form-inline">
+            <div className="pf-c-input-group">
+              <input placeholder="30s"
+                type="text" className="pf-c-form-control"
```

Comment:
Nit: `className` attribute should be on its own line.
(and same for the `input`s below)